devicetypes/smartthings/zigbee-window-shade.src/zigbee-window-shade.groovy

```diff
@@ -82,113 +83,159 @@ private List<Map> collectAttributes(Map descMap) {
-        cmd = zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_GOTO_LIFT_PERCENTAGE, zigbee.convertToHexString(data, 2))
-    } else {
-        cmd = zigbee.command(zigbee.LEVEL_CONTROL_CLUSTER, COMMAND_MOVE_LEVEL_ONOFF, zigbee.convertToHexString(Math.round(data * 255 / 100), 2))
-    }
+	log.info "setLevel()"
+	def cmd
+	if (supportsLiftPercentage()) {
+		if(isIkeaKadrilj()){
+			// Ikea Kadrilj keeps % level of being closed (instead of % level of being opened)
+			// inverting that logic is needed here
+			def newData = 100 - data
+			cmd = zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_GOTO_LIFT_PERCENTAGE, zigbee.convertToHexString(newData, 2))
+		} else {
+			cmd = zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_GOTO_LIFT_PERCENTAGE, zigbee.convertToHexString(data, 2))
+		}
+
+	} else {
+		cmd = zigbee.command(zigbee.LEVEL_CONTROL_CLUSTER, COMMAND_MOVE_LEVEL_ONOFF, zigbee.convertToHexString(Math.round(data * 255 / 100), 2))
+	}
 
-    return cmd
+	return cmd
 }
 
 def pause() {
-    log.info "pause()"
-    zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_PAUSE)
+	log.info "pause()"
+	zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_PAUSE)
 }
 
 /**
  * PING is used by Device-Watch in attempt to reach the Device
  * */
 def ping() {
-    return refresh()
+	return refresh()
 }
 
 def refresh() {
-    log.info "refresh()"
-    def cmds
-    if (supportsLiftPercentage()) {
-        cmds = zigbee.readAttribute(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT)
-    } else {
-        cmds = zigbee.readAttribute(zigbee.LEVEL_CONTROL_CLUSTER, ATTRIBUTE_CURRENT_LEVEL)
-    }
-    return cmds
+	log.info "refresh()"
+	def cmds
+	if (supportsLiftPercentage()) {
+		cmds = zigbee.readAttribute(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT)
+	} else {
+		cmds = zigbee.readAttribute(zigbee.LEVEL_CONTROL_CLUSTER, ATTRIBUTE_CURRENT_LEVEL)
+	}
+	return cmds
 }
 
 def configure() {
-    // Device-Watch allows 2 check-in misses from device + ping (plus 2 min lag time)
-    log.info "configure()"
-    sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
-    log.debug "Configuring Reporting and Bindings."
-
-    def cmds
-    if (supportsLiftPercentage()) {
-        cmds = zigbee.configureReporting(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT, DataType.UINT8, 0, 600, null)
-    } else {
-        cmds = zigbee.levelConfig()
-    }
-    return refresh() + cmds
+	// Device-Watch allows 2 check-in misses from device + ping (plus 2 min lag time)
+	log.info "configure()"
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+	log.debug "Configuring Reporting and Bindings."
+
+	def cmds
+	if (supportsLiftPercentage()) {
+		cmds = zigbee.configureReporting(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT, DataType.UINT8, 0, 600, null)
+		if(isIkeaKadrilj()){
```

Comment:
Similar to my last comment, I suggest creating a second function called something like `configReadBindingTable` (or preferably a better name than that)

Reply:
ok, done